src/ecephys_etl/modules/align_timestamps/probe_synchronizer.py

```diff
@@ -0,0 +1,164 @@
+        ----------
+        samples : numpy.ndarray
+            Array of timestamps in probe samples.
+        sync_condition : str, optional
+            How to synchronize the timestamps. Available options are:
+                'master': Default, synchronize to master clock
+                'probe': adjust probe samples -> probe times
+
+        Returns
+        -------
+        numpy.ndarray : 
+            Sample timestamps in seconds on the master (default) or probe clock.
+
+        """
+
+        in_range = np.where(
+            ((samples / self.local_probe_sampling_rate) >= self.min_time)
+            * ((samples / self.local_probe_sampling_rate) < self.max_time)
+        )[0]
+
+        if self.global_probe_sampling_rate > 0:
+
+            if sync_condition == "probe":
+                samples[in_range] = samples[in_range] / self.local_probe_sampling_rate
+
+            elif sync_condition == "master":
+                samples[in_range] = (
+                    samples[in_range] / self.global_probe_sampling_rate
+                    - self.total_time_shift
+                )
+
+            else:
+                raise ValueError(
+                    "unrecognized sync condition: {}".format(sync_condition)
+                )
+
+        else:
+            samples[in_range] = -1
+
+        return samples
+
+    @classmethod
+    def compute(
+        cls,
+        master_barcode_times,
+        master_barcodes,
+        probe_barcode_times,
+        probe_barcodes,
+        min_time,
+        max_time,
+        probe_start_index,
+        local_probe_sampling_rate,
+    ):
+        """Compute a transform from probe samples to master times by aligning barcodes.
+
+        Parameters
+        ----------
+        master_barcode_times : np.ndarray
+            start times of barcodes (according to the master clock) on the master line. 
+            One per barcode.
+        master_barcodes : np.ndarray
+            barcode values on the master line. One per barcode
+        probe_barcode_times : np.ndarray
+            start times (according to the probe clock) of barcodes on the probe line. 
+            One per barcode
+        probe_barcodes : np.ndarray
+            barcode values on the probe_line. One per barcode
+        min_time : Float
+            time (in seconds) of first barcode to align
+        max_time : Float
+            time (in seconds) of last barcode to align
+        probe_start_index : int
+            sample index of probe acquisition start time
+        local_probe_sampling_rate : float
+            the probe's apparent sampling rate
+    
+        Returns
+        -------
+        ProbeSynchronizer : 
+            When called, applies the transform computed here to samples on the probe clock.
+
+        """
+
+        times_array = np.array(probe_barcode_times)
+        barcodes_array = np.array(probe_barcodes)
+
+        ok_barcodes = np.where((times_array > min_time) * (times_array < max_time))[0]
```

Comment:
Can you make this an `np.logical_and`? Multiplying arrays of booleans seems unnecessarily convoluted.